ddl/ddl_api.go

```diff
@@ -1628,20 +1628,20 @@ func checkInvisibleIndexOnPK(tblInfo *model.TableInfo) error {
 	if tblInfo.PKIsHandle {
 		return nil
 	}
-	pk := getPrimaryKey(tblInfo)
+	pk := GetPrimaryKey(tblInfo)
 	if pk != nil && pk.Invisible {
 		return dbterror.ErrPKIndexCantBeInvisible
 	}
 	return nil
 }
 
-// getPrimaryKey extract the primary key in a table and return `IndexInfo`
+// GetPrimaryKey extract the primary key in a table and return `IndexInfo`
 // The returned primary key could be explicit or implicit.
 // If there is no explicit primary key in table,
 // the first UNIQUE INDEX on NOT NULL columns will be the implicit primary key.
 // For more information about implicit primary key, see
 // https://dev.mysql.com/doc/refman/8.0/en/invisible-indexes.html
-func getPrimaryKey(tblInfo *model.TableInfo) *model.IndexInfo {
+func GetPrimaryKey(tblInfo *model.TableInfo) *model.IndexInfo {
```

Comment:
Why does ```GetPrimaryKey``` become public ？

Reply:
Hi, wanna reuse it in TiFlow (cdc). Will it cause problems?  ref: https://github.com/pingcap/tiflow/issues/5902

Reply:
OK, I think ```GetPrimaryKey``` can become the number function of ```model.TableInfo```.

Reply:
make sense